src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathSingleFilter.java

```diff
@@ -0,0 +1,124 @@
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.filters;
+
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.TreeWalkerAuditEvent;
+import com.puppycrawl.tools.checkstyle.TreeWalkerFilter;
+import com.puppycrawl.tools.checkstyle.api.AutomaticBean;
+
+/**
+ * This filter processes {@link TreeWalkerAuditEvent}
+ * objects based on the criteria of files, checks, message, module id, xpathQuery.
+ */
+public class SuppressionXpathSingleFilter extends AutomaticBean implements
+        TreeWalkerFilter {
+    /**
+     * XpathFilter instance.
+     */
+    private XpathFilter xpathFilter;
+    /**
+     * The pattern for file names.
+     */
+    private Pattern files;
+    /**
+     * The pattern for check class names.
+     */
+    private Pattern checks;
+    /**
+     * The pattern for message names.
+     */
+    private Pattern message;
+    /**
+     * Module id of filter.
+     */
+    private String id;
+    /**
+     * Xpath query.
+     */
+    private String query;
+
+    /**
+     * Set the regular expression to specify names of files to suppress.
+     * @param files the name of the file
+     */
+    public void setFiles(String files) {
+        if (files == null) {
+            this.files = null;
+        }
+        else {
+            this.files = Pattern.compile(files);
+        }
+    }
+
+    /**
+     * Set the regular expression to specify the name of the check to suppress.
+     * @param checks the name of the check
+     */
+    public void setChecks(String checks) {
+        if (checks == null) {
+            this.checks = null;
+        }
+        else {
+            this.checks = Pattern.compile(checks);
+        }
+    }
+
+    /**
+     * Set the regular expression to specify the message of the check to suppress.
+     * @param message the message of the check
+     */
+    public void setMessage(String message) {
+        if (message == null) {
+            this.message = null;
+        }
+        else {
+            this.message = Pattern.compile(message);
+        }
+    }
+
+    /**
+     * Set the ID of the check to suppress.
+     * @param id the ID of the check
+     */
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    /**
+     * Set the xpath query.
+     * @param query the xpath query
+     */
+    public void setQuery(String query) {
+        this.query = query;
+    }
+
+    @Override
+    public boolean accept(TreeWalkerAuditEvent treeWalkerAuditEvent) {
+        return xpathFilter.accept(treeWalkerAuditEvent);
+    }
+
+    @Override
+    protected void finishLocalSetup() {
+        xpathFilter = new XpathFilter(files, checks, message, id, query);
+    }
```

Comment:
please move this method above `accept`.
It is good to keep method in class in order of their execution from outside.

Reply:
done.